projects/storefrontlib/src/cms-components/asm/services/asm-enabler.service.ts

```diff
@@ -79,7 +83,7 @@ export class AsmEnablerService {
     const factory = this.componentFactoryResolver.resolveComponentFactory(
       AsmMainUiComponent
     );
-    this.outletService.add('cx-storefront', factory, OutletPosition.BEFORE);
+    this.triggerService.renderDialog(this.triggerConfig.trigger?.asm, factory);
```

Comment:
I'd prefer to delegate the creation of the factory to the triggerService (or other generic service):
- save us from a constructor arg in each caller service
- we can do it in a generic fashion
- we can potentially leverage lazy-loading capabilities, sproutlets, etc. 

In order to do it right, we should align ourselves with the generic loading of components.
Let's align with Stan on this one.